core/src/main/kotlin/cash/grammar/kotlindsl/utils/DependencyExtractor.kt

```diff
@@ -57,6 +57,28 @@ public class DependencyExtractor(
       .asContainer()
   }
 
+  /**
+   * Given that we're inside a `dependencies {}` block, collect the set of dependencies with their context.
+   */
+  public fun collectDependencyDeclarationsWithContext(ctx: NamedBlockContext): List<Pair<DependencyDeclaration, PostfixUnaryExpressionContext>> {
```

Comment:
@autonomousapps 

For context, I need access to the `PostfixUnaryExpressionContext` to manipulate it, as shown in [this example](https://github.com/autonomousapps/dependency-analysis-gradle-plugin/pull/1299/files#diff-b2a94f740ad8ff6516bf944965d212e085314087d0edc9cc852710ad1ef35c21R119-R125).

I considered adding it as a property of `DependencyDeclaration`, but that would couple ANTLR with the parsed result and make testing more difficult. Instead, I decided to create a separate function to handle this.

Keen to hear if you have another suggestion on how we should implement this.

Reply:
No, I don't think we want to do it like this. That means two APIs that basically do the same thing, one of which isn't actually useful (the original one). Let's figure out a way to continue to use the original `collectDependencies()` and include the necessary information in the `DependencyContainer`. While I was working on that class yesterday, I realized it needs better modeling. I'm not sure what that looks like yet. Maybe `DependencyDeclaration` should include the `StatementContext` it is parsed from, maybe not.

Hmm thinking out loud... maybe `DependencyDeclaration` needs a higher-level abstraction with two "views" -- one of the views is the current declaration model class, and the other view is just the `StatementContext`. That keeps the current unit tests working, while also providing access to the necessary underlying "primitive" (the `StatementContext`).

One idea that might help is that `DependencyContainer.elements` is built in an ordered way. We are simply iterating over each statement in the dependencies block and parsing them as best we can, adding them to the container one by one, in order. So maybe we could include an internal index, and two primary internal properties. One is the parsed statements, and one is the raw contexts. We can have methods that get the elements as parsed models or as raw contexts. Dunno, again, just brainstorming.

Reply:
As we discussed offline, I have removed this function and made the existing `collectDependencies` to return DependencyContainer with either (1) `DependencyDeclarationWithContext` or (2) StatementContext. 4ee2bccaca89d77598546f035c4e1534f1c98a87